cmd/federation-controller/main.go

```diff
@@ -205,19 +215,41 @@ func main() {
 		}
 	}
 
-	istioClient, err := istiokube.NewClient(istiokube.NewClientConfigForRestConfig(kubeConfig), "")
-	if err != nil {
-		log.Fatalf("failed to create Istio client: %v", err)
-	}
-
-	rm := kube.NewReconcilerManager(
-		meshConfigPushRequests,
+	reconcilers := []kube.Reconciler{
 		kube.NewGatewayResourceReconciler(istioClient, istioConfigFactory),
 		kube.NewServiceEntryReconciler(istioClient, istioConfigFactory),
 		kube.NewWorkloadEntryReconciler(istioClient, istioConfigFactory),
-		kube.NewVirtualServiceReconciler(istioClient, istioConfigFactory),
-		kube.NewDestinationRuleReconciler(istioClient, istioConfigFactory),
-	)
+		kube.NewPeerAuthResourceReconciler(istioClient, namespace),
+	}
+	if cfg.MeshPeers.Local.IngressType == config.OpenShiftRouter {
+		routeClient, err := routev1client.NewForConfig(kubeConfig)
+		if err != nil {
+			log.Fatalf("failed to create Route client: %v", err)
+		}
+
+		reconcilers = append(reconcilers, kube.NewDestinationRuleReconciler(istioClient, istioConfigFactory))
+		reconcilers = append(reconcilers, kube.NewEnvoyFilterReconciler(istioClient, istioConfigFactory))
+		reconcilers = append(reconcilers, kube.NewRouteReconciler(routeClient, openshift.NewConfigFactory(*cfg, serviceLister)))
+
+		go func() {
+			log.Debugf("Resolving %s", cfg.MeshPeers.Remote.Addresses[0])
+			lastIPs := networking.Resolve(cfg.MeshPeers.Remote.Addresses[0])
```

Comment:
I know this looks better and that was my first approach, but this results in the log message: "IP addresses of %s have changed". And I thought it's better to not print it on the initial resolution, because if I as a user will want to check if the DNS record changes over time and triggers reconciliation I will see the message even though the IPs were not really changed.

Reply:
Makes sense. Thanks.